Model/src/main/java/org/gusdb/wdk/model/user/dataset/event/UserDatasetEventListHandler.java

```diff
@@ -41,20 +41,26 @@ public static void main(String[] args) {
   @Override
   protected void execute() throws Exception {
     var projectId = (String) getOptionValue(ARG_PROJECT);
-    var handler   = new UserDatasetEventArrayHandler(projectId);
-    var eventFile = new File((String)getOptionValue(ARG_EVENTS_FILE));
 
-    JSONArray eventJsonArray;
-    try (FileInputStream fileInputStream = new FileInputStream(eventFile)) {
-      eventJsonArray = new JSONArray(fileInputStream.toString());
+    switch ((String) getOptionValue(ARG_RUN_MODE)) {
+      case "cleanup":
+        new UserDatasetEventCleanup(projectId).cleanupFailedInstalls();
```

Comment:
Yes, until we have a better process around this.  We decided to continue to require manual investigation of event handling errors.  In practice this means that when an error email is sent out, someone will need to figure out what went wrong and if it can be corrected.  Once the issue is corrected, they will move the event from the `failed` status to the `cleanup_ready` status.

Reply:
Yep, after investigating the event failure and fixing whatever the cause was, the row must be manually changed to CLEANUP_READY

Reply:
It's totally workable, but I want to mention it's not just failed uploads - sometimes a user might notice their dataset didn't get correctly installed due to a bug, and then an operator might want to clean up a successful upload - or possibly a few.


How much potential does the operator have for messing up? Say there's a UD with a successful install, a share, an uninstall, and another install. Assume the uninstaller is idempotent (uninstalling twice is the same as uninstalling once). Does it matter which events exactly they set to CLEANUP_READY?